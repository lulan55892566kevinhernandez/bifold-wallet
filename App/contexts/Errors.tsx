import React, { useState, useContext } from 'react'
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native'

import AppStyles from '../../assets/styles'

const ErrorsContext = React.createContext({})

interface IErrorDialog {
  path: string
  text: string
}

function ErrorDialog(props: IErrorDialog) {
  const errors = useContext<any>(ErrorsContext)

  return (
    <>
      <View style={styles.errorView}>
        <View style={[AppStyles.messageBox, styles.errorMessage]}>
          <Text style={[AppStyles.h2, AppStyles.textSecondary, AppStyles.textBold]}>ERROR{'\n'}</Text>
          <Text style={[AppStyles.h3, AppStyles.textSecondary]}>{props.text}</Text>
          <TouchableOpacity
            style={[AppStyles.button, AppStyles.backgroundSecondary, { marginTop: 30 }]}
            onPress={() => {
              errors.setVisible(false)
            }}
          >
            <Text style={[AppStyles.h2, AppStyles.textWhite]}>Okay</Text>
          </TouchableOpacity>
        </View>
      </View>
    </>
  )
}

interface Props {
  children: React.ReactNode
}

const Errors: React.FC<Props> = ({ children }) => {
  const [visible, setVisible] = useState(false)
  const [text, setText] = useState('')
  const [path, setPath] = useState('')

  return (
    <ErrorsContext.Provider value={{ setVisible: setVisible, setText: setText, setPath: setPath }}>
      <View>
        {children}
        {visible ? (
          <>
            <ErrorDialog text={text} path={path} />
          </>
        ) : null}
      </View>
    </ErrorsContext.Provider>
  )
}

export default Errors
export { ErrorsContext }

const styles = StyleSheet.create({
  errorView: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    alignSelf: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgba(200, 200, 200, 0.78)',
  },
  errorMessage: {
    bottom: 40,
    width: '84%',
    height: '44%',
  },
})
