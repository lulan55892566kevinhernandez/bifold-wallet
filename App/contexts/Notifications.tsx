import React, { useState, useContext } from 'react'
import { Image, Text, TouchableOpacity, View, StyleSheet } from 'react-native'

import Images from '../../assets/images'

const NotificationsContext = React.createContext({})

interface INotification {
  title: string
  text: string
  image: string
  path: string
}

function Notification(props: INotification) {
  const notifications = useContext<any>(NotificationsContext)

  return (
    <View style={styles.msgView}>
      <View style={styles.innerView}>
        <TouchableOpacity
          onPress={() => {
            notifications.setVisible(false)
          }}
        >
          {props.title ? (
            <Text style={[AppStyles.h2, AppStyles.textSecondary, AppStyles.textBold]}>
              {props.title}
              {'\n'}
            </Text>
          ) : null}
          {props.text ? (
            <Text style={AppStyles.textSecondary}>
              {props.text}
              {'\n'}
            </Text>
          ) : null}
          {props.image ? <Image source={Images.arrow} style={{ alignSelf: 'center', width: 60, height: 68 }} /> : null}
        </TouchableOpacity>
      </View>
    </View>
  )
}

interface INotifications {
  children: JSX.Element[]
}

function Notifications(props: INotifications) {
  const [visible, setVisible] = useState(false)
  const [text, setText] = useState('')
  const [path, setPath] = useState('')
  const [title, setTitle] = useState('')
  const [image, setImage] = useState('')

  return (
    <NotificationsContext.Provider
      value={{
        setVisible: setVisible,
        setText: setText,
        setPath: setPath,
        setTitle: setTitle,
        setImage: setImage,
      }}
    >
      {props.children}
      {visible ? <Notification text={text} path={path} title={title} image={image} /> : null}
    </NotificationsContext.Provider>
  )
}

export default Notifications
export { NotificationsContext }

const styles = StyleSheet.create({
  msgView: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    alignSelf: 'center',
    backgroundColor: 'rgba(200, 200, 200, 0.78)',
  },
  innerView: {
    width: '100%',
    height: '83%',
    top: '17%',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderTopLeftRadius: 50,
    borderTopRightRadius: 50,
    padding: 50,
  },
})
