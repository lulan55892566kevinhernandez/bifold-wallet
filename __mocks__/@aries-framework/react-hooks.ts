import { CredentialRecord, ProofRecord } from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ credentials: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofRecord[])
const mockAgent = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
}
const useAgent = () => ({
  agent: {
    credentials: mockAgent,
  },
})
const useCredentialById = jest.fn()

export { useAgent, useCredentialById, useCredentials, useCredentialByState, useProofByState }
