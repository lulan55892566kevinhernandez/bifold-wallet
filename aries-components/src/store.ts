import { Agent } from 'aries-framework'
import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux'

import { agentSlice } from './slices/agent'
import { connectionsSlice } from './slices/connections/connectionsSlice'

const rootReducer = combineReducers({
  agent: agentSlice.reducer,
  connections: connectionsSlice.reducer,
})

type RootState = ReturnType<typeof rootReducer>
const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

const initializeStore = (agent: Agent) => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        thunk: {
          extraArgument: {
            agent,
          },
        },
      }),
  })

  type AppDispatch = typeof store.dispatch
  const useAppDispatch = () => useDispatch<AppDispatch>()

  return {
    store,
    useAppDispatch,
  }
}

export { initializeStore, useAppSelector }

export type { RootState }
