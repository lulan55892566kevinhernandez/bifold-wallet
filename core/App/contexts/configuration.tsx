import { createContext, useContext } from 'react'

import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { Overlay } from '../types/overlay'

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  OCABundle: { branding: Record<string, Overlay>; oca: JSON | undefined }
  useBiometry: React.FC
  record: React.FC<RecordProps>
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
