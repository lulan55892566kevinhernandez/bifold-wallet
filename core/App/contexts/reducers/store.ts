import AsyncStorage from '@react-native-async-storage/async-storage'

import { LocalStorageKeys } from '../../constants'
import {
  Preferences as PreferencesState,
  Onboarding as OnboardingState,
  Authentication as AuthenticationState,
  Lockout as LockoutState,
  LoginAttempt as LoginAttemptState,
  State,
} from '../../types/state'

enum OnboardingDispatchAction {
  ONBOARDING_UPDATED = 'onboarding/onboardingStateLoaded',
  DID_COMPLETE_TUTORIAL = 'onboarding/didCompleteTutorial',
  DID_AGREE_TO_TERMS = 'onboarding/didAgreeToTerms',
  DID_CREATE_PIN = 'onboarding/didCreatePIN',
}

enum LockoutDispatchAction {
  LOCKOUT_UPDATED = 'lockout/lockoutUpdated',
}

enum LoginAttemptDispatchAction {
  ATTEMPT_UPDATED = 'loginAttempt/loginAttemptUpdated',
}

enum PreferencesDispatchAction {
  ENABLE_DEVELOPER_MODE = 'preferences/enableDeveloperMode',
  USE_BIOMETRY = 'preferences/useBiometry',
  PREFERENCES_UPDATED = 'preferences/preferencesStateLoaded',
}

enum AuthenticationDispatchAction {
  DID_AUTHENTICATE = 'authentication/didAuthenticate',
}

enum DeepLinkDispatchAction {
  ACTIVE_DEEP_LINK = 'deepLink/activeDeepLink',
}

export type DispatchAction =
  | OnboardingDispatchAction
  | LoginAttemptDispatchAction
  | LockoutDispatchAction
  | PreferencesDispatchAction
  | AuthenticationDispatchAction
  | DeepLinkDispatchAction

export const DispatchAction = {
  ...OnboardingDispatchAction,
  ...LoginAttemptDispatchAction,
  ...LockoutDispatchAction,
  ...PreferencesDispatchAction,
  ...AuthenticationDispatchAction,
  ...DeepLinkDispatchAction,
}

export interface ReducerAction<R> {
  type: R
  payload?: Array<any>
}

export const reducer = <S extends State>(state: S, action: ReducerAction<DispatchAction>): S => {
  switch (action.type) {
    case PreferencesDispatchAction.ENABLE_DEVELOPER_MODE: {
      const choice = (action?.payload ?? []).pop() ?? false
      const preferences = { ...state.preferences, developerModeEnabled: choice }

      AsyncStorage.setItem(LocalStorageKeys.Preferences, JSON.stringify(preferences))

      return {
        ...state,
        preferences,
      }
    }
    case PreferencesDispatchAction.USE_BIOMETRY: {
      const choice = (action?.payload ?? []).pop() ?? false
      const preferences = {
        ...state.preferences,
        useBiometry: choice,
      }
      const onboarding = {
        ...state.onboarding,
        didConsiderBiometry: true,
      }
      const newState = {
        ...state,
        onboarding,
        preferences,
      }

      AsyncStorage.setItem(LocalStorageKeys.Onboarding, JSON.stringify(onboarding))
      AsyncStorage.setItem(LocalStorageKeys.Preferences, JSON.stringify(preferences))

      return newState
    }
    case PreferencesDispatchAction.PREFERENCES_UPDATED: {
      const preferences: PreferencesState = (action?.payload || []).pop()
      return {
        ...state,
        preferences,
      }
    }
    case LoginAttemptDispatchAction.ATTEMPT_UPDATED: {
      const loginAttempt: LoginAttemptState = (action?.payload || []).pop()
      const newState = {
        ...state,
        loginAttempt,
      }
      AsyncStorage.setItem(LocalStorageKeys.LoginAttempts, JSON.stringify(newState.loginAttempt))
      return newState
    }
    case LockoutDispatchAction.LOCKOUT_UPDATED: {
      const lockout: LockoutState = (action?.payload || []).pop()
      return {
        ...state,
        lockout,
      }
    }
    case OnboardingDispatchAction.ONBOARDING_UPDATED: {
      const onboarding: OnboardingState = (action?.payload || []).pop()
      return {
        ...state,
        onboarding,
      }
    }
    case OnboardingDispatchAction.DID_COMPLETE_TUTORIAL: {
      const onboarding = {
        ...state.onboarding,
        didCompleteTutorial: true,
      }
      const newState = {
        ...state,
        onboarding,
      }
      AsyncStorage.setItem(LocalStorageKeys.Onboarding, JSON.stringify(newState.onboarding))
      return newState
    }
    case OnboardingDispatchAction.DID_AGREE_TO_TERMS: {
      const onboarding: OnboardingState = {
        ...state.onboarding,
        didAgreeToTerms: true,
      }
      const newState = {
        ...state,
        onboarding,
      }
      AsyncStorage.setItem(LocalStorageKeys.Onboarding, JSON.stringify(newState.onboarding))
      return newState
    }
    case OnboardingDispatchAction.DID_CREATE_PIN: {
      const onboarding: OnboardingState = {
        ...state.onboarding,
        didCreatePIN: true,
      }
      const newState = {
        ...state,
        onboarding,
      }
      AsyncStorage.setItem(LocalStorageKeys.Onboarding, JSON.stringify(newState.onboarding))
      return newState
    }
    case AuthenticationDispatchAction.DID_AUTHENTICATE: {
      const value: AuthenticationState = (action?.payload || []).pop()
      const payload = value ?? { didAuthenticate: true }
      const newState = {
        ...state,
        ...{ authentication: payload },
      }
      return newState
    }
    case DeepLinkDispatchAction.ACTIVE_DEEP_LINK: {
      const value = (action?.payload || []).pop()
      return {
        ...state,
        ...{ deepLink: { activeDeepLink: value } },
      }
    }
    default:
      return state
  }
}

export default reducer
