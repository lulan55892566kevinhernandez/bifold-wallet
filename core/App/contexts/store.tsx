import React, { createContext, Dispatch, useContext, useReducer } from 'react'

import { State } from '../types/state'

import reducer, { ReducerAction } from './reducers/store'

const initialState: State = {
  onboarding: {
    didAgreeToTerms: false,
    didCompleteTutorial: false,
    didCreatePIN: false,
    didConsiderBiometry: false,
  },
  authentication: {
    didAuthenticate: false,
  },
  lockout: {
    displayNotification: false,
  },
  privacy: {
    didShowCameraDisclosure: false,
  },
  preferences: {
    useBiometry: false,
  },
  credential: {
    revoked: new Set(),
    revokedMessageDismissed: new Set(),
  },
  error: null,
  loading: false,
}

export const StoreContext = createContext<[State, Dispatch<ReducerAction>]>([
  initialState,
  () => {
    return
  },
])

export const StoreProvider: React.FC = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return <StoreContext.Provider value={[state, dispatch]}>{children}</StoreContext.Provider>
}

export const useStore = () => useContext(StoreContext)
