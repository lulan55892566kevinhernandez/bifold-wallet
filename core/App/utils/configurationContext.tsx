import { createContext, useContext } from 'react'

import { pages } from '../screens/OnboardingPages'

export interface ConfigurationContext {
  onboarding: {
    pages: typeof pages
  }
  splash: React.FC
  terms: React.FC
}

const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)
export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfigurationContext = () => useContext(ConfigurationContext)
