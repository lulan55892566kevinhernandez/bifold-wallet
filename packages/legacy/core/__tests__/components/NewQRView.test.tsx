import { useConnections } from '@aries-framework/react-hooks'
import { render } from '@testing-library/react-native'
import React from 'react'

import NewQRView from '../../App/components/misc/NewQRView'
import { ConfigurationContext } from '../../App/contexts/configuration'
import { StoreProvider, defaultState } from '../../App/contexts/store'
import configurationContext from '../contexts/configuration'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('NewQRView Component', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    // @ts-ignore
    useConnections.mockReturnValue({ records: [] })
  })

  test('Renders correctly on first tab', async () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <NewQRView defaultToConnect={false} handleCodeScan={() => Promise.resolve()} />
      </ConfigurationContext.Provider>
    )

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly on second tab', async () => {
    const tree = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            developerModeEnabled: false,
            biometryPreferencesUpdated: false,
            useBiometry: false,
            useVerifierCapability: false,
            useConnectionInviterCapability: false,
            useDevVerifierTemplates: false,
            walletName: 'My Wallet - 1234',
          },
        }}
      >
        <ConfigurationContext.Provider value={configurationContext}>
          <NewQRView defaultToConnect={true} handleCodeScan={() => Promise.resolve()} />
        </ConfigurationContext.Provider>
      </StoreProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
