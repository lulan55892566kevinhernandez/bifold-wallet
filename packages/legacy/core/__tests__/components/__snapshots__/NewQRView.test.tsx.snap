// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQRView Component Renders correctly on first tab 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#313132",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Scan.WillScanAutomatically
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 24,
            "borderWidth": 2,
            "height": 250,
            "width": 250,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="Scan.Torch"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderColor": "#FFFFFF",
          "borderRadius": 24,
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "marginBottom": 50,
          "opacity": 1,
          "width": 24,
        }
      }
      testID="com.ariesbifold:id/ScanTorch"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#313132",
        "borderTopWidth": 0,
        "flexDirection": "row",
        "height": 60,
        "paddingBottom": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      }
    }
  >
    <View
      accessibilityLabel="Scan.ScanQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.ScanQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="Scan.MyQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.MyQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`NewQRView Component Renders correctly on second tab 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#313132",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
            testID="com.ariesbifold:id/LoadingActivityIndicator"
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": "33%",
                  "width": "33%",
                }
              }
              testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
            />
            <View
              collapsable={false}
              style={
                Object {
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "360deg",
                    },
                  ],
                }
              }
            >
              <
                fill="#FFFFFF"
                height={200}
                width={200}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  Object {
                    "paddingHorizontal": 20,
                  },
                ]
              }
              testID="com.ariesbifold:id/WalletName"
            >
              My Wallet - 1234
            </Text>
            <View
              accessibilityLabel="NameWallet.EditWalletName"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                Object {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 44,
                  "left": 44,
                  "right": 44,
                  "top": 44,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="com.ariesbifold:id/EditWalletName"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginLeft": 0,
                    "marginRight": 15,
                    "minHeight": 26,
                    "minWidth": 26,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 26,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰏫
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
                "marginTop": 20,
                "textAlign": "center",
              }
            }
          >
            Connection.ShareQR
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#313132",
        "borderTopWidth": 0,
        "flexDirection": "row",
        "height": 60,
        "paddingBottom": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      }
    }
  >
    <View
      accessibilityLabel="Scan.ScanQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.ScanQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="Scan.MyQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.MyQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
